File: src/BaseHmiTypes/Screens/Shapes/HmiShapeBase.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Shapes;

public abstract class HmiShapeBase : HmiSimpleScreenItemBase
{
    public HmiProperty<HmiColor>? LineColor { get; set; }

    public HmiProperty<HmiColor>? AlternateLineColor { get; set; }

    public HmiProperty<double>? LineWidth { get; set; }

    public HmiProperty<int>? DashType { get; set; }

    public HmiProperty<int>? BackFillPattern { get; set; }

    public HmiProperty<HmiFillPattern>? FillPattern { get; set; }

    public HmiProperty<HmiColor>? TransparentColor { get; set; }

    public HmiProperty<bool>? UseTransparentColor { get; set; }

    public HmiProperty<HmiColor>? FirstGradientColor { get; set; }

    public HmiProperty<double>? FirstGradientOffset { get; set; }

    public HmiProperty<HmiColor>? MiddleGradientColor { get; set; }

    public HmiProperty<HmiColor>? SecondGradientColor { get; set; }

    public HmiProperty<double>? SecondGradientOffset { get; set; }

    public HmiProperty<bool>? UseFirstGradient { get; set; }

    public HmiProperty<bool>? UseSecondGradient { get; set; }

    public HmiProperty<bool>? UseDesignColorSchema { get; set; }

    public HmiProperty<bool>? UseDesignShadowSettings { get; set; }
}

File: typescript/base-hmi-types/src/screens/shapes/HmiShapeBase.ts
import { HmiColor, HmiProperty } from "../base.js";
import { HmiFillPattern } from "../base/HmiFillPattern.js";
import { HmiSimpleScreenItemBase } from "../base/HmiSimpleScreenItemBase.js";

export abstract class HmiShapeBase extends HmiSimpleScreenItemBase {
  lineColor?: HmiProperty<HmiColor>;
  alternateLineColor?: HmiProperty<HmiColor>;
  lineWidth?: HmiProperty<number>;
  dashType?: HmiProperty<number>;
  backFillPattern?: HmiProperty<number>;
  fillPattern?: HmiProperty<HmiFillPattern>;
  transparentColor?: HmiProperty<HmiColor>;
  useTransparentColor?: HmiProperty<boolean>;
  firstGradientColor?: HmiProperty<HmiColor>;
  firstGradientOffset?: HmiProperty<number>;
  middleGradientColor?: HmiProperty<HmiColor>;
  secondGradientColor?: HmiProperty<HmiColor>;
  secondGradientOffset?: HmiProperty<number>;
  useFirstGradient?: HmiProperty<boolean>;
  useSecondGradient?: HmiProperty<boolean>;
  useDesignColorSchema?: HmiProperty<boolean>;
  useDesignShadowSettings?: HmiProperty<boolean>;
}
